Modules/Sources/JetpackStats/Analytics/StatsTracker+ErrorTracking.swift

```diff
@@ -0,0 +1,55 @@
+import Foundation
+
+extension StatsTracker {
+    /// Convenience method to track errors with automatic type detection
+    /// - Parameters:
+    ///   - error: The error to track
+    ///   - screen: The screen where the error occurred
+    func trackError(_ error: Error, screen: String) {
+        let errorType: String
+        let errorCode = (error as NSError).code
+
+        // Determine error type based on the error instance
+        switch error {
+        case let urlError as URLError:
+            errorType = urlErrorType(urlError)
+        case is DecodingError:
+            errorType = "parsing"
+        case is CancellationError:
+            return
+        default:
+            // Check for common error domains
+            let nsError = error as NSError
+            switch nsError.domain {
+            case NSCocoaErrorDomain:
+                errorType = "cocoa_\(errorCode)"
+            case NSURLErrorDomain:
+                errorType = "url_\(errorCode)"
+            default:
+                errorType = "unknown"
+            }
+        }
```

Comment:
I'm not sure if it's worth using this block of code to create an "error type". `(error as NSError).domain` is probably enough?